lib/commands/device/move.ts

```diff
@@ -145,42 +152,56 @@ export default class DeviceMoveCmd extends Command {
 		balena: BalenaSDK,
 		devices: ExtendedDevice[],
 	) {
-		const [deviceDeviceTypes, deviceTypes] = await Promise.all([
-			Promise.all(
-				devices.map((device) =>
-					balena.models.device.getManifestBySlug(
-						device.is_of__device_type[0].slug,
-					),
+		const { getExpandedProp } = await import('../../utils/pine');
+		// deduplicate the slugs
+		const deviceCpuArchs = Array.from(
+			new Set(
+				devices.map(
+					(d) => d.is_of__device_type[0].is_of__cpu_architecture[0].slug,
 				),
 			),
-			balena.models.config.getDeviceTypes(),
-		]);
-
-		const compatibleDeviceTypes = deviceTypes.filter((dt) =>
-			deviceDeviceTypes.every(
-				(deviceDeviceType) =>
-					balena.models.os.isArchitectureCompatibleWith(
-						deviceDeviceType.arch,
-						dt.arch,
-					) &&
-					!!dt.isDependent === !!deviceDeviceType.isDependent &&
```

Comment:
Decided to drop this check since:
* it's not part of the model yet
* dependent devices isn't released as a feature
* only users that know how to enable the dependent devices feature & that already have a dependent app of generic DT will see any difference